packages/cli-kit/src/public/common/string.ts

```diff
@@ -113,3 +114,87 @@ export function getRandomName(family: RandomNameFamily = 'business'): string {
   }
   return `${takeRandomFromArray(mapping[family].adjectives)}-${takeRandomFromArray(mapping[family].nouns)}`
 }
+
+/**
+ * Given a string, it returns it with the first letter capitalized
+ */
+export function capitalize(str: string): string {
+  return str.substring(0, 1).toUpperCase() + str.substring(1)
+}
+
+/**
+ * Try to convert a string to an int, falling back to undefined if unable to
+ */
+export function tryParseInt(maybeInt: string | undefined): number | undefined {
+  let asInt: number | undefined
+  if (maybeInt !== undefined) {
+    asInt = parseInt(maybeInt, 10)
+    if (isNaN(asInt)) {
+      asInt = undefined
+    }
+  }
+  return asInt
+}
+
+/**
+ * Given a series of rows inside an array, where each row is an array of strings (representing columns)
+ * Parse it into a single string with the columns aligned
+ */
+export function linesToColumns(lines: string[][]): string {
+  const widths: number[] = []
+  for (let i = 0; lines[0] && i < lines[0].length; i++) {
+    const columnRows = lines.map((line) => line[i]!)
+    widths.push(Math.max(...columnRows.map((row) => unstyled(row).length)))
+  }
+  const paddedLines = lines
+    .map((line) => {
+      return line
+        .map((col, index) => {
+          return `${col}${' '.repeat(widths[index]! - unstyled(col).length)}`
+        })
+        .join('   ')
+        .trimEnd()
+    })
+    .join('\n')
+  return paddedLines
+}
+
+/**
+ * Given a string, it transforms it to a slug (lowercase, hyphenated, no special chars, trimmed...)
+ */
+export function slugify(str: string): string {
+  return str
+    .toLowerCase()
+    .trim()
+    .replace(/[^\w\s-]/g, '')
+    .replace(/[\s_-]+/g, '-')
+    .replace(/^-+|-+$/g, '')
+}
+
+/**
+ * Given a string, it returns it with the special regex characters escaped
+ * More info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+ */
+export function escapeRegExp(str: string): string {
```

Comment:
Nitpick: Even though it's common to see functions in Javascript that are not very idiomatic. I think we can use this refactoring opportunity to make ours more idiomatic. Ruby does a better job at this with conventions to indicate whether you are modifying the input argument or returning a new copy using the `!` suffix. What about naming the functions here:

```
import {
  slugifiedString,
  camelCasedString,
  paramCasedString,
  snakeCasedString,
  constantCasedString,
  capitalizedString,
  decodeIntegerFromString
} from "@shopify/cli-kit/common/string"
```

Reply:
Hmm, I'm not sure to see the benefit. In JS the parameters are always passed by value (well, the simple ones at least), so why do we need to make a distinction?

For example, I think `camelize(myString)` is quite clear and simple. I know it will just generate a new string because it can't modify my variable. Also, we are following this style (infinitive) in many other places.